### PR TITLE
Make `initialization` a required field in `azuredevops_git_repository`

### PR DESCRIPTION
**Pull request by beandrad**
Instead of crashing when no initialization type is provided.

## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [] ~~I have updated the documentation accordingly~~ -> Documentation already implies behavior.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number: #54

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by EliiseS** [comment]
@xuzhang3 Should `initalized `

File: azuredevops/internal/service/git/resource_git_repository.go [hunk 1]
```diff
@@ -394,9 +394,14 @@ func expandGitRepository(d *schema.ResourceData) (*git.GitRepository, *repoIniti
 
 	var initialization *repoInitializationMeta = nil
 	initData := d.Get("initialization").(*schema.Set).List()
-
-	// Note: If configured, this will be of length 1 based on the schema definition above.
-	if len(initData) == 1 {
+	if len(initData) == 0 {
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository.go]: @xuzhang3 @tmeckel Should initialization be a required field? I can't find a example where it's not set to be something,

Comment (xuzhang3) [reply azuredevops/internal/service/git/resource_git_repository.go]: Not required. But if not set, terraform will crash #54

Comment (beandrad) [reply azuredevops/internal/service/git/resource_git_repository.go]: This PR should fix that and default to `Uninitialized`. I think `Uninitialized` makes sense as a default, as the repo will be created with the default branch and no commits.

Comment (EliiseS) [reply azuredevops/internal/service/git/resource_git_repository.go]: @beandrad Instead initializing it with default, let's make the `initialization` field required with 

```go
			"initialization": {
				Type:     schema.TypeSet,
				Required: true,
				MaxItems: 1,
```

Comment (tmeckel) [reply azuredevops/internal/service/git/resource_git_repository.go]: I agree with @EliiseS , better force people to be explicit on what they want, instead for heading with a (potentially) unwanted default.

File: azuredevops/internal/service/git/resource_git_repository_test.go [hunk 2]
```diff
@@ -121,7 +122,7 @@ func TestGitRepo_FlattenExpand_RoundTrip(t *testing.T) {
 	require.Equal(t, *expandedGitRepo.Id, repoID)
 	require.NotNil(t, expandedProjectID)
 	require.Equal(t, *expandedProjectID, projectID)
-	require.Nil(t, repoInitialization)
+	require.Equal(t, repoInitialization.initType, string(RepoInitTypeValues.Clean))
 }
 
 func TestGitRepo_FlattenExpandInitialization_RoundTrip(t *testing.T) {
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository_test.go]: Is this test testing anything different from the `TestGitRepo_FlattenExpand_RoundTrip` now? 

Maybe this? But could you move these to the above test?
```go
	require.Equal(t, repoInitialization.sourceType, "")
	require.Equal(t, repoInitialization.sourceURL, "")
```

Also, having the `configureCleanInitialization(resourceData)` come after the `flattenRepository`, seems incorrect. That wouldn't happen ever in a normal use case would it? @tmeckel

Comment (beandrad) [reply azuredevops/internal/service/git/resource_git_repository_test.go]: oh yeah, they seem to be the same now; I guess I can just remove `TestGitRepo_FlattenExpand_RoundTrip` and rename `TestGitRepo_FlattenExpandInitialization_RoundTrip` as `TestGitRepo_FlattenExpand_RoundTrip`.

Comment (tmeckel) [reply azuredevops/internal/service/git/resource_git_repository_test.go]: @EliiseS your're right the `flattenRepository` must come last! It's purpose is to serialize all settings into the Terraform state and if you change settings after this serialization the changes won't end up in the state.

File: azuredevops/internal/service/git/resource_git_repository_test.go [hunk 3]
```diff
@@ -109,29 +109,6 @@ func TestGitRepo_FlattenExpand_RoundTrip(t *testing.T) {
...
-	projectID := uuid.New()
-	project := core.TeamProjectReference{Id: &projectID}
-
-	repoID := uuid.New()
-	repoName := "name"
-	gitRepo := git.GitRepository{Id: &repoID, Name: &repoName, Project: &project}
-
 	resourceData := schema.TestResourceDataRaw(t, ResourceGitRepository().Schema, nil)
 	resourceData.SetId(gitRepo.Id.String())
 	flattenGitRepository(resourceData, &gitRepo)
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository_test.go]: Can you switch this line with the one below it?

**Comment by EliiseS** [comment]
Looks great! Thank you so much for your contribution!
Ship it 😄 
```
     _
    /|\
   /_|_\
 ____|____
 \_o_o_o_/
~~ |     ~~~~~
___t_________
````

**Pull request merged**